src/test/java/emissary/test/core/junit5/UnitTest.java

```diff
@@ -199,7 +225,19 @@ protected Document getAnswerDocumentFor(String resource) {
             return null;
         }
 
-        String aname = resource.substring(0, datPos) + ResourceReader.XML_SUFFIX;
+        String aname = "";
+        if (answerFiles.isEmpty()) {
+            aname = resource.substring(0, datPos) + ResourceReader.XML_SUFFIX;
+        } else {
+            // if answer files are in different directory than data files, this will be used to find matching answer to data pair
+            String testFileName = FilenameUtils.getBaseName(resource);
+            for (String answer : answerFiles) {
+                if (FilenameUtils.getBaseName(answer).equals(testFileName)) {
+                    aname = answer;
```

Comment:
Suggest we break out after finding a match.
```suggestion
                    aname = answer;
                    break;
```